docs/theory-of-operation.md

```diff
@@ -0,0 +1,208 @@
+
+
+- **Enable/Disable Network/PXE boot and press `n` for iPXE boot string**
+
+  This feature lets user `Enable/Disable` booting from PXE. If it is `Enabled`
+  then during boot process user can access PXE boot string with `n` button. Also
+  appropriate boot device should be available in boot menu (e.g. `iPXE`).
+
+  Watch [Video](https://asciinema.org/a/241583) showing how to enable/disable
+  PXE boot and how to open PXE boot string.
+
+
+- **Enable/Disable USB boot**
+
+	This feature let user `Enable/Disable` booting from USB device.
+
+  Watch [Video](https://asciinema.org/a/239796) showing how to enable/disable
+  USB boot.
+
+
+- **Disable serial console and enable with S1 button**
+
+  To `Disable` serial console type `t` in payload menu. If it is disabled no
+  data will be displayed in serial output. After changing state to `Disabled`
+  restoring console can be done only via S1 button.
+
+  Watch [video](https://asciinema.org/a/240951) showing how to disable serial
+  console with payload menu option and then enable it with S1 button.
+
+
+- **Redirect console output to COM2**
+
+  Console output is available via COM1 in default. This feature let user
+  redirect console output to COM2.
+  >NOTE: Changing this option from `Disabled` to `Enabled` (or the opposite way)
+  will cause losing current output in terminal. The advice is to open new
+  connection in another terminal window on another COM.
+
+  Watch [video](https://asciinema.org/a/240926) showing how to redirect console
+  output to COM2.
+  >NOTE: In above example serial connection is redirect to telnet. First, it is
+  opened on 13541 port (COM1). After enabling redirect console to COM2, output
+  is no longer available. New telnet connection is opened on 13542 port (COM2)
+  and console output is available there.
+
+
+- **Enable/Disable CPU boost**
+
+  This feature lets user `Enable/Disable` CPU performance boost. To verify if
+  it works, memory test can be done (in boot menu choose `payload[memtest]`).
+
+  >NOTE: Notice how memory transfer speed changes depending on CPU boost
+  enable/disable.
+
+  Watch [video](https://asciinema.org/a/241571) showing how to enable/disable
+  CPU performance boost and verify it.
+
+
+- **Enable/Disable UARTC/UARTD**
+
+	This feature lets user enable/disable superIO UARTx on GPIO header.  
+
+	Watch [Video](https://asciinema.org/a/241576) showing how to enable/disable
+	UARTC/UARTD.
+
+
+- **Enable/Disable mPCIe2 clk**
+
+  If mPCIe2 clk is enabled then GPP3 PCIe clock (which is attached to apu2
+  mPCIe2 slot) is always on. It is used when extension card is attached to
+  mPCIe2 slot. If no extension card is attached it is advised to set to
+  `Disable`.
+
+  Watch [Video](https://asciinema.org/a/239813) showing how to enable/disable
+  mPCIe2 clk.
+
+
+- **Enable/Disable EHCI0 controller**
+
+	This feature lets user enable/disable EHCI0 controller.
+
+	Watch [Video](https://asciinema.org/a/244779) showing how to enable/disable
```

Comment:
@piotr-kleins please grep the lspci output to get only the EHCI controllers. Displaying whole output is not eye-friendly. How am I supposed to see any change in such huge list of PCI devices.

Reply:
I show that feature on apu5 and list only USB controllers now.

Reply:
@piotr-kleins great. One more thing. The description and title of the video does not mention it is PC Engines apu. please edit the title and description in he asciinema site.

Reply:
Done for all videos.